aleph-client/src/treasury.rs

```diff
@@ -0,0 +1,152 @@
+use crate::{try_send_xt, wait_for_event, AnyConnection, RootConnection, SignedConnection};
+use ac_primitives::ExtrinsicParams;
+use codec::Decode;
+use frame_support::PalletId;
+use primitives::Balance;
+use sp_core::{Pair, H256};
+use sp_runtime::{traits::AccountIdConversion, AccountId32};
+use std::{thread, thread::sleep, time::Duration};
+use substrate_api_client::{compose_extrinsic, ApiResult, GenericAddress, XtStatus};
+
+type AnyResult<T> = anyhow::Result<T>;
+
+/// Returns the account of the treasury.
+pub fn treasury_account() -> AccountId32 {
+    PalletId(*b"a0/trsry").into_account_truncating()
+}
+
+/// Returns how many treasury proposals have ever been created.
+///
+/// Requires a single storage read.
+pub fn proposals_counter<C: AnyConnection>(connection: &C) -> u32 {
+    connection
+        .as_connection()
+        .get_storage_value("Treasury", "ProposalCount", None)
+        .expect("Key `Treasury::ProposalCount` should be present in storage")
+        .unwrap_or(0)
+}
+
+/// Calculates how much balance will be paid out to the treasury after each era.
+///
+/// Every call to this method is potentially expensive - it requires three storage reads.
```

Comment:
Depends how we look at it - those get key-value pair RPC calls are not that expensive as e.g. sending  extrinsic that requires more encoding usually. Making this comment only because other methods in this file does not have such comment. If we take metric of how many TCP calls we need to do under the hood then I agree 3 requests are more than 1 :)

Reply:
yeah, you are right, I was trying to be kinda informative, but it was silly; removed